pkg/metrics/utils.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"sort"
+
+	v1 "github.com/leptonai/gpud/api/v1"
+	"github.com/leptonai/gpud/components"
+	components_metrics_state "github.com/leptonai/gpud/pkg/gpud-metrics/state"
+)
+
+func ConvertToLeptonMetrics(ms Metrics) v1.LeptonMetrics {
+	aggregated := make(map[string][]components.Metric)
+	for _, m := range ms {
+		if m.Component == "" {
+			continue
+		}
+		if _, ok := aggregated[m.Component]; !ok {
+			aggregated[m.Component] = make([]components.Metric, 0)
+		}
+
+		aggregated[m.Component] = append(aggregated[m.Component], components.Metric{
+			Metric: components_metrics_state.Metric{
```

Comment:
The dependency between components.Metric and components_metrics_state.Metric is a bit weird.  IMO components.Metric is a basic type, it can rely on some types defined in components package but not other packages.  We can improve it in the future.

Reply:
We will remove this once we migrate all to this PR